### PR TITLE
add tagging for upload.bufferToUssFile

### PR DESCRIPTION
**Pull request by jace-roell**
**What It Does**
Add file tagging to `Upload.bufferToUssFile`

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Review by ATorrise — APPROVED** [review]

**Review by anaxceron — COMMENTED** [review]
Requesting an edit to the changelog : )

File: packages/zosfiles/__tests__/__unit__/methods/upload/Upload.unit.test.ts [hunk 1]
```diff
@@ -1661,6 +1661,9 @@ describe("z/OS Files - Upload", () => {
             expect(zosmfExpectSpy).toHaveBeenCalledWith(dummySession, { reqHeaders: headers, resource: endpoint, writeData: data });
         });
         it("should return with proper response when upload USS file in binary", async () => {
+            const chtagSpy = jest.spyOn(Utilities, "chtag");
```
Comment (zFernand0) [inline packages/zosfiles/__tests__/__unit__/methods/upload/Upload.unit.test.ts:1664]: should we `expect(thisSpy).toHaveBeenCalled()`?

File: packages/zosfiles/src/methods/upload/Upload.ts [hunk 2]
```diff
@@ -487,6 +488,12 @@ export class Upload {
             apiResponse.etag = uploadRequest.response.headers.etag;
         }
 
+        if (options.encoding != null) {
+            await Utilities.chtag(session, origUssname, Tag.TEXT, options.encoding);
```
Comment (zFernand0) [inline packages/zosfiles/src/methods/upload/Upload.ts:492]: it seems that this line is not covered?

Do we not have any tests that call the `bufferToUssFile` with no encoding? 🤯 
If not, maybe we should modify/add one 😋

File: packages/zosfiles/src/methods/upload/Upload.ts [hunk 3]
```diff
@@ -487,6 +488,12 @@ export class Upload {
...
+        } else if (options.binary) {
+            await Utilities.chtag(session, origUssname, Tag.BINARY);
+        }
+
         return {
             success: true,
             commandResponse: ZosFilesMessages.dataSetUploadedSuccessfully.message,
```
Comment (zFernand0) [inline packages/zosfiles/src/methods/upload/Upload.ts:499]: also, not sure if we should make this change... but the apiResponse is empty 😋 
Should we at least specify which file the buffer was uploaded to? and maybe the contents of the buffer too? 😅 

image

File: packages/zosfiles/src/methods/upload/Upload.ts [hunk 4]
```diff
@@ -487,6 +488,12 @@ export class Upload {
             apiResponse.etag = uploadRequest.response.headers.etag;
         }
 
+        if (options.encoding != null) {
+            await Utilities.chtag(session, origUssname, Tag.TEXT, options.encoding);
+        } else if (options.binary) {
+            await Utilities.chtag(session, origUssname, Tag.BINARY);
```
Comment (zFernand0) [inline packages/zosfiles/src/methods/upload/Upload.ts:494]: I'm afraid that the file on the mainframe isn't tagged after testing.


```
- untagged    T=off -rw-r--r--    1 FERNANDO 0           4163 Dec  6 15:30 test-file1.txt
```
I could be doing something wrong though.

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Took me a few minutes to figure out testing steps since `Upload.bufferToUssFile` is not exposed on any CLI command 😓 

Still testing since I ran into a small hiccup 😋 


<details><summary>Full testing script</summary>

```typescript
/* Save this file as `test.mjs` and run `node test.mjs` */

import { ProfileInfo } from "@zowe/imperative";
import { Upload } from "@zowe/zos-files-for-zowe-sdk";
import * as fs from "fs";
(async () => {
    // Load connection info from default z/OSMF profile
    const profInfo = new ProfileInfo("zowe");
    await profInfo.readProfilesFromDisk();
    const zosmfProfAttrs = profInfo.getDefaultProfile("zosmf");
    const zosmfMergedArgs = profInfo.mergeArgsForProfile(zosmfProfAttrs, { getSecureVals: true });
    const session = ProfileInfo.createSession(zosmfMergedArgs.knownArgs);

    const localFile = "package.json";
    const localFileContents = fs.readFileSync(localFile).toString();
    const remoteLocation = "/u/users/fernando/test-file1.txt";
    const options = { binary: true };
    // const response = await Upload.fileToUssFile(session, localFile, remoteLocation, options);
    const response = await Upload.bufferToUssFile(session, remoteLocation, Buffer.from(localFileContents), options);
    console.log(response);
})().catch((err) => {
    console.error(err);
    process.exit(1);
});
```
</details>

**Review by awharn — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 

When I tested it, I had a version of master cached in the build.
My mistake.
Files get tagged as expected 🙏

**Comment by zFernand0** [comment]
Besides the changelog updates requested by Ana, I think that my comments are mostly nice-to-have(s).
Don't feel obligated to implement them 😋 
Just figured I mention it to have other your thoughts on them 😋

**Comment by zFernand0** [comment]
The sonar cloud issues are related to very old commented out code.
Feel free to remove those commented out lines of code if you would like, but I don't believe that it should hold the PR from being merged 😋 
image
image

File: packages/zosfiles/src/methods/upload/Upload.ts (hunk 3 above)
Comment (zFernand0) [reply packages/zosfiles/src/methods/upload/Upload.ts:499]: Curious about your thoughts on this, @jace-roell?

File: packages/zosfiles/src/methods/upload/Upload.ts (hunk 2 above)
Comment (zFernand0) [reply packages/zosfiles/src/methods/upload/Upload.ts:492]: Hey @jace-roell,
was this resolved in some un-pushed commit?

File: packages/zosfiles/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe z/OS files SDK package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Corrected `Upload.BufferToUssFile()` SDK function to properly tag uploaded files. [#2378](https://github.com/zowe/zowe-cli/pull/2378)
```
Comment (anaxceron) [inline packages/zosfiles/CHANGELOG.md:7]: can we add "the" here?: "Corrected **the** `bbbb` SDK function to ... ."

**Review by anaxceron — APPROVED** [review]
Requesting a minor edit to the changelog

**Review by gejohnston — APPROVED** [review]

**Pull request merged**